osu.Game/Online/Chat/MessageNotifier.cs

```diff
@@ -0,0 +1,257 @@
+
+                if (notifyOnMention.Value && anyCaseInsensitive(words, localUsername))
+                {
+                    var notification = new MentionNotification(message.Sender.Username, onClick);
+                    notificationOverlay?.Post(notification);
+
+                    continue;
+                }
+
+                if (!string.IsNullOrWhiteSpace(highlightWords.Value))
+                {
+                    var matchedWord = hasCaseInsensitive(words, getWords(highlightWords.Value));
+
+                    if (matchedWord != null)
+                    {
+                        var notification = new HighlightNotification(message.Sender.Username, matchedWord, onClick);
+                        notificationOverlay?.Post(notification);
+                    }
+                }
+            }
+        }
+
+        private static string[] getWords(string input) => input.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
+
+        /// <summary>
+        /// Finds the first matching string/word in both <paramref name="x"/> and <paramref name="y"/> (case-insensitive)
+        /// </summary>
+        private static string hasCaseInsensitive(IEnumerable<string> x, IEnumerable<string> y) => x.FirstOrDefault(x2 => anyCaseInsensitive(y, x2));
+
+        private static bool anyCaseInsensitive(IEnumerable<string> x, string y) => x.Any(x2 => x2.Equals(y, StringComparison.OrdinalIgnoreCase));
+
+        public class HighlightNotification : SimpleNotification
+        {
+            public HighlightNotification(string highlighter, string word, Action onClick)
+            {
+                Icon = FontAwesome.Solid.Highlighter;
+                Text = $"'{word}' was mentioned in chat by '{highlighter}'. Click to find out why!";
+                this.onClick = onClick;
+            }
+
+            private readonly Action onClick;
+
+            public override bool IsImportant => false;
+
+            [BackgroundDependencyLoader]
+            private void load(OsuColour colours)
+            {
+                IconBackgound.Colour = colours.PurpleDark;
+                Activated = delegate
+                {
+                    onClick?.Invoke();
+                    return true;
+                };
+            }
+        }
+
+        public class PrivateMessageNotification : SimpleNotification
+        {
+            public PrivateMessageNotification(string username, Action onClick)
+            {
+                Icon = FontAwesome.Solid.Envelope;
+                Username = username;
+                MessageCount = 1;
+                this.onClick = onClick;
+            }
+
+            private int messageCount;
+
+            public int MessageCount
+            {
+                get => messageCount;
+                set => Text = (messageCount = value) > 1 ? $"You received {messageCount} private messages from '{Username}'. Click to read it!" : $"You received a private message from '{Username}'. Click to read it!";
```

Comment:
I wouldn't benefit from Humanizer because it still requires me to put a condition in place (e.g. `messageCount > 1 ? "message".Pluralize() : "message"`).

Reply:
Nevermind. https://github.com/Humanizr/Humanizer#toquantity